packages/oauth/src/index.ts

```diff
@@ -312,30 +319,43 @@ export class InstallProvider {
     req: IncomingMessage,
     res: ServerResponse,
     options?: CallbackOptions,
+    installOptions?: InstallURLOptions,
   ): Promise<void> {
     let parsedUrl;
     let code: string;
     let flowError: string;
     let state: string;
-    let installOptions: InstallURLOptions;
-
     try {
       if (req.url !== undefined) {
-        parsedUrl = new URL(req.url);
+        parsedUrl = new URL(req.url, `http://${req.headers.host}`);
```

Comment:
The new recommended way to url parse via URL constructor is less lenient about what it will accept than the previous url.parse(). So partial urls don't work any more.

Reply:
The URL should be `https://`. We don't allow using `http://` in production OAuth flows. Also, does this approach work well in `X-Forwarded-Host` header situations too?

Reply:
Hey @seratch! OK is it fair to say that when the request contains x-forwarded-host, we would always prefer to take the host value from there?

Reply:
@srajiang If you would like to know the actual redirect URI here, you need to use `x-forwarded-` headers. Speaking of this part, whether this line of code works well also for the situation depends on how the Node's `URL` module behaves. Before going with this implementation, we should do testing with a reverse proxy. It seems that we are using only `searchParams` so far. As long as the URL initialization does not fail, it should work, though.

By the way, is this code taken from Node's official recommendation?

Reply:
I agree that we should do testing with reverse proxy; I would want to know how the `x-forwarded-` header is  accessed from the `req` object.  

Re: Is this code taken from Node's offical recommendation - I'm not sure exactly what you mean. I've been referencing [these docs](https://nodejs.org/api/url.html#url_new_url_input_base). Based on that, it seems that if we provide the correc original host, it should work.

Maybe I am missing something here?

Reply:
Thanks. As `http://${req.headers.host}` is not a good-enough solution, I was curious about the reason why I chose the way. To properly build the `base` part of the `URL` constructor arguments, you can use the pair of `x-forwarded-proto` and `x-forwarded-host` in the case where they exist in `req`. Perhaps, we want to have an internal method that builds the `base` from `req` object.

Reply:
@srajiang and I discussed the best way to implement this and we decided to have a private method that returns only searchParams from `req` as we don't use any other properties in `req` here.

Reply:
BTW for context on why the URL assembly changed: it was part of the move to ESLint. ESLint will warn (via the [`node/no-deprecated-api` rule](https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-deprecated-api.md)) if you use the [now-deprecated node method `require('url').parse`](https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost). ESLint instead recommends to construct the URL using the `new` keyword.